src/api/plugins/tdbp/warren_tdbp/indicators.py

```diff
@@ -0,0 +1,259 @@
+"""Warren TdBP indicators."""
+
+from datetime import datetime, timedelta
+from typing import List, Optional
+
+import pandas as pd
+from functools import cached_property
+from pydantic import HttpUrl
+from ralph.backends.http import LRSHTTP
+from ralph.backends.http.lrs import LRSQuery
+from ralph.conf import LRSHeaders
+from warren.backends import lrs_client as async_lrs_client
+from warren.filters import DatetimeRange
+from warren.indicators import BaseIndicator
+from warren.xi import ExperienceRead
+from warren.xi.clients import CRUDExperience as async_xi_experience_client
+
+from .models import ActiveAction, SlidingWindow, StudentScore
+
+
+class SlidingWindowIndicator(BaseIndicator):
+    """TODO"""
+
+    course_experiences: List[str] = None
+    sliding_window: DatetimeRange = None
+    active_actions: List[ActiveAction] = None
+    dynamic_cohort: List[str] = None
+
+    def __init__(  # noqa:PLR0913
+        self,
+        course_id: str,
+        until: Optional[datetime] = None,
+        sliding_window_size: int = 15,
+        active_actions_min: int = 6,
+        dynamic_cohort_size: int = 3,
```

Comment:
We need to define defaults for those parameters (and make them configurable as for warren settings):

```suggestion
        # from . import settings
        sliding_window_size: int = settings.SLIDING_WINDOW_SIZE,
        active_actions_min: int = settings.ACTIVE_ACTIONS_MIN,
        dynamic_cohort_size: int = settings.DYNAMIC_COHORT_SIZE,
```